File: Ejercicio 18/ejercicio18.py
class Pregunta:
    def __init__(self, enunciado):
        self.enunciado = enunciado

class Examen:
    def __init__(self, materia):
        self.materia = materia
        self.preguntas = []

    # Composición: Instanciación interna
    def agregar_pregunta(self, texto):
        nueva_pregunta = Pregunta(texto)
        self.preguntas.append(nueva_pregunta)

# --- PRUEBA ---
if __name__ == "__main__":
    evaluacion = Examen("Inteligencia Artificial")
    
    evaluacion.agregar_pregunta("Defina Redes Neuronales.")
    evaluacion.agregar_pregunta("Explique el aprendizaje supervisado.")

    print(f"Evaluación: {evaluacion.materia}")
    for idx, p in enumerate(evaluacion.preguntas, 1):
        print(f"{idx}. {p.enunciado}")

File: Ejercicio 18/ejercicio18.java
import java.util.ArrayList;
import java.util.List;

class Pregunta {
    private String enunciado;

    public Pregunta(String enunciado) {
        this.enunciado = enunciado;
    }

    public String getEnunciado() {
        return enunciado;
    }
}

class Examen {
    private String materia;
    private List<Pregunta> preguntas;

    public Examen(String materia) {
        this.materia = materia;
        this.preguntas = new ArrayList<>();
    }

  
    public void añadirPregunta(String texto) {
        this.preguntas.add(new Pregunta(texto));
    }

    public void mostrarExamen() {
        System.out.println("Examen de: " + materia);
        for (int i = 0; i < preguntas.size(); i++) {
            System.out.println((i + 1) + ". " + preguntas.get(i).getEnunciado());
        }
    }
}

public class Main {
    public static void main(String[] args) {
        Examen parcial = new Examen("Fundamentos de Software");
        
      
        parcial.añadirPregunta("¿Qué es la composición en POO?");
        parcial.añadirPregunta("Diferencia entre Java y Python en memoria.");

        parcial.mostrarExamen();
    }
}
